packages/frontend/tests/unit/service/channels/vote-verifier-test.ts

```diff
@@ -95,73 +83,361 @@ module('Unit | Service | channels/vote-verifier', function (hooks) {
         previousVoteChain: secondVote,
         signature: undefined,
       });
-      currentVote.signature = await sign(
-        await hash(generateSortedVote(currentVote)),
-        user3.privateSigningKey
-      );
+      currentVote.signature = await signatureOf(currentVote, user3);
 
-      assert.ok(await service.verify(currentVote));
+      assert.ok(await service.isValid(currentVote));
     });
   });
 
   module('vote is not valid', function () {
-    test('when there are many chains', async function (assert) {
-      const store = getStore();
+    module('when there is a single chain', function () {
+      test('when first voter is in both yes and no', async function (assert) {
+        const store = getStore();
+        let user1 = await buildUser('user1');
+        let userToAdd = await buildUser('userToAdd');
 
-      let user1 = await buildUser('user1');
-      let user2 = await buildUser('user2');
-      let user3 = await buildUser('user3');
-      let user4 = await buildUser('user4');
-      let userToAdd = await buildUser('userToAdd');
+        let currentVote = store.createRecord('vote-chain', {
+          yes: [user1],
+          no: [user1],
+          remaining: [],
+          action: VOTE_ACTION.ADD,
+          target: userToAdd,
+          key: user1,
+          previousVoteChain: undefined,
+          signature: undefined,
+        });
+        currentVote.signature = await signatureOf(currentVote, user1);
 
-      let firstVote = store.createRecord('vote-chain', {
-        yes: [user1],
-        no: [],
-        remaining: [user2, user3, user4],
-        action: VOTE_ACTION.ADD,
-        target: userToAdd,
-        key: user1,
-        previousVoteChain: undefined,
-        signature: undefined,
+        assert.notOk(await service.isValid(currentVote));
       });
-      firstVote.signature = await sign(
-        await hash(generateSortedVote(firstVote)),
-        user1.privateSigningKey
-      );
 
-      let secondVote = store.createRecord('vote-chain', {
-        yes: [user1, user2],
-        no: [],
-        remaining: [user3, user4],
-        action: VOTE_ACTION.ADD,
-        target: userToAdd,
-        key: user2,
-        previousVoteChain: firstVote,
-        signature: undefined,
+      test('when first voter is still in remaining', async function (assert) {
```

Comment:
rather than saying "in remaining" (which is in implementation detail), what does it mean to be in remaining?
An integration-style test name should describe behavior more than a particular implementation